src/mozillavpn.cpp

```diff
@@ -712,6 +713,14 @@ void MozillaVPN::serversFetched(const QByteArray& serverData) {
   }
 }
 
+void MozillaVPN::subscriptionDetailsFetched(
+    const QByteArray& subscriptionDetailsData) {
+  logger.debug() << "Subscription details data fetched!";
+  Q_UNUSED(subscriptionDetailsData);
+
+  requestSubscriptionManagement();
```

Comment:
Does the loading view have a cancel button/link? if yes, we don't know if we are in the right state here

Reply:
Plus, why not calling directly `emit subscriptionManagementNeeded()`? You can get rid of `requestSubscriptionManagement` it seem.